### ISSUE TITLE
Add pyiqa as local scoring backend

### ISSUE THREAD

**Issue by francisbrero** [francisbrero/photo-scoring#23]
## Description
Integrate the [pyiqa](https://github.com/chaofengc/IQA-PyTorch) library as a local scoring option, enabling free, fast, GPU-accelerated image quality assessment without API calls.

## Background
Research into similar projects revealed pyiqa as the most comprehensive open-source IQA toolbox, with 30+ metrics including NIMA, MUSIQ, TOPIQ, and BRISQUE.

## Benefits
- **Zero cost** - No API fees for local scoring
- **Speed** - GPU accelerated, can process 1000+ images quickly
- **Privacy** - Images never leave the machine
- **Offline** - Works without internet connection

### Installation
```bash
pip install pyiqa
```

### Basic Usage
```python
import pyiqa
import torch

# Create aesthetic scorer (NIMA)
nima = pyiqa.create_metric('nima', device=torch.device('cuda'))

# Create technical scorer (MUSIQ or TOPIQ)
musiq = pyiqa.create_metric('musiq', device=torch.device('cuda'))

# Score image
aesthetic_score = nima('path/to/image.jpg')
technical_score = musiq('path/to/image.jpg')
```

### Available Metrics to Consider
| Metric | Type | Notes |
|--------|------|-------|
| `nima` | Aesthetic | Google's Neural Image Assessment |
| `musiq` | Both | Multi-scale transformer, any resolution |
| `topiq_nr` | Both | State-of-art no-reference metric |
| `brisque` | Technical | Classic blind quality metric |
| `niqe` | Technical | Natural image quality evaluator |

## Tasks
- [ ] Add pyiqa to optional dependencies
- [ ] Create `LocalScoringClient` class mirroring `OpenRouterClient` interface
- [ ] Map pyiqa scores (typically 1-10 or 0-100) to our 0-1 scale
- [ ] Add `--provider local` CLI flag
- [ ] Benchmark speed vs API (images/second)
- [ ] Compare score correlation with our VLM scores
- [ ] Document GPU requirements

## Hybrid Approach (Future)
Consider using local scoring for fast pre-filtering, then VLM critique only for top candidates:
1. Score all images locally with pyiqa (fast, free)
2. Filter to top 20% by score
3. Run VLM critique on filtered set (detailed feedback)

## References
- [pyiqa GitHub](https://github.com/chaofengc/IQA-PyTorch)
- [NIMA Paper](https://arxiv.org/abs/1709.05424)
- [MUSIQ Paper](https://arxiv.org/abs/2108.05997)
